File: cases/taint/python/model-store-separation-negative/store.py
class Store:
    def put(self, key, value):
        pass

    def get(self, key):
        return ""


alpha = Store()
beta = Store()


def dfb_source():  # DFB-SOURCE: model-store-separation-input
    return "tainted"


def dfb_sink(value):  # DFB-SINK: model-store-separation-sink
    pass


def write_side():
    alpha.put("k", dfb_source())


def read_side():
    dfb_sink(beta.get("k"))

File: cases/taint/javascript/model-store-separation-positive/Store.js
class Store {
  put(key, value) {}

  get(key) {
    return "";
  }
}

const alpha = new Store();
const beta = new Store();

function dfb_source() { // DFB-SOURCE: model-store-separation-input
  return "tainted";
}

function dfb_sink(value) {} // DFB-SINK: model-store-separation-sink

function writeSide() {
  alpha.put("k", dfb_source());
}

function readSide() {
  dfb_sink(alpha.get("k"));
}
